arkouda/pdarrayclass.py

```diff
@@ -920,10 +921,139 @@ def save(self, prefix_path, dataset='array', mode='truncate') -> str:
         return generic_msg("tohdf {} {} {} {} {}".\
                            format(self.name, dataset, m, json_array, self.dtype))
 
+
+    def register(self, user_defined_name : str) -> 'pdarray':
+        """
+        Return a pdarray with a user defined name in the arkouda server 
+        so it can be attached to later using pdarray.attach()
+        
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name array is to be registered under
+        
+        Returns
+        -------
+        pdarray
+            pdarray which points to original input pdarray but is also 
+            registered with user defined name in the arkouda server
+        
+        Raises
+        ------
+        TypeError
+            Raised if pda is neither a pdarray nor a str or if 
+            user_defined_name is not a str
+        
+        See also
+        --------
+        attach, unregister
+        
+        Notes
+        -----
+        Registered names/pdarrays in the server are immune to deletion 
+        until they are unregistered.
+        
+        Examples
+        --------
+        >>> a = zeros(100)
+        >>> r_pda = a.register("my_zeros")
+        >>> # potentially disconnect from server and reconnect to server
+        >>> b = ak.pdarray.attach("my_zeros")
+        >>> # ...other work...
+        >>> b.unregister()
+        """
+        return register_pdarray(self, user_defined_name)
+
+    def unregister(self) -> None:
+        """
+        Unregister a pdarray in the arkouda server which was previously 
+        registered using register() and/or attahced to using attach()
+        
+        Parameters
+        ----------
+        user_defined_name : str
+            which array was registered under
+        
+        Returns
+        -------
+        None
+        
+        Raises 
+        ------
+        TypeError
+            Raised if pda is neither a pdarray nor a str
+        
+        See also
+        --------
+        register, unregister
+        
+        Notes
+        -----
+        Registered names/pdarrays in the server are immune to deletion until 
+        they are unregistered.
+        
+        Examples
+        --------
+        >>> a = zeros(100)
+        >>> r_pda = a.register("my_zeros")
+        >>> # potentially disconnect from server and reconnect to server
+        >>> b = ak.pdarray.attach("my_zeros")
+        >>> # ...other work...
+        >>> b.unregister()
+        """
+        unregister_pdarray(self)
+        
+    # class method self is not passed in
+    # invoke with ak.pdarray.attach('user_defined_name')
+    @staticmethod
```

Comment:
recommend adding typechecked annotation

Reply:
this brings up a question i had since the method being called is `@typechecked`

do you need it because there are many places where other functions which are `@typechecked` are immediately called